core/workspace.js

```diff
@@ -490,8 +491,8 @@ Blockly.Workspace.prototype.fireChangeListener = function(event) {
  */
 Blockly.Workspace.prototype.getBlockById = function(id) {
   var block = this.blockDB_[id];
-  if (!block && this.getFlyout() && this.getFlyout().getWorkspace()) {
-    block = this.getFlyout().getWorkspace().blockDB_[id];
+  if (!block && this.getFlyout_() && this.getFlyout_().getWorkspace()) {
```

Comment:
this.getFlyout_ will be undefined if this is an unrendered workspace.

Either leave getFlyout defined in Blockly.Workspace (and rename it to getFlyout_) or move this definition of getBlockById into WorkspaceSvg and make the Workspace definition a simple lookup in the blockDB_.

You probably aren't getting bitten by this right now because you're not using headless scratch-blocks.  I don't have a strong opinion on whether you should fix it in this merge or in a separate PR--if the latter, please make an issue about it.